### PR TITLE
theme implementation

### PR DESCRIPTION
**Pull request by Pony315**

### CODE REVIEW

File: src/app/topic/TopicsChart.tsx [hunk 1]
```diff
@@ -63,9 +63,28 @@ function TopicsChart() {
...
+              },
+              '&:active': {
+                  backgroundColor: 'info.light', 
+              },
+              '&:disabled': {
+                  color: 'grey.500',
+                  borderColor: 'grey.500',
+                  backgroundColor: 'primary.light',
+              }
+          }}
```
Comment (timicienio) [inline src/app/topic/TopicsChart.tsx:83]: 這裡應該用某種方式讓 CSS 可以重用

File: src/app/topic/TopicsChart.tsx [hunk 2]
```diff
@@ -75,9 +94,26 @@ function TopicsChart() {
...
+              },
+              '&:active': {
+                  backgroundColor: 'info.dark', 
+              },
+              '&:disabled': {
+                  color: 'grey.500',
+                  borderColor: 'grey.300',
+                  backgroundColor: 'grey.300',
+              }
+          }}
```
Comment (timicienio) [inline src/app/topic/TopicsChart.tsx:116]: Same here

File: src/app/topic/[topicId]/Posts.tsx [hunk 3]
```diff
@@ -43,16 +41,14 @@ export default function Posts({
   };
 
   const { data, isLoading } = useTopicPosts(topicId, {
-    limit: 12,
-    offset: (page - 1) * 12,
```
Comment (timicienio) [inline src/app/topic/[topicId]/Posts.tsx:47]: 不要把 limit offset 寫死

File: src/app/topic/[topicId]/Posts.tsx [hunk 4]
```diff
@@ -114,7 +108,7 @@ export default function Posts({
             <CircularProgress color="inherit" />
           </Box>
         )}
-        {data?.data.posts.map((post) => (
+         {data?.data.posts.slice((page - 1) * 10, page * 10).map((post) => (
```
Comment (timicienio) [inline src/app/topic/[topicId]/Posts.tsx:111]: 不要這樣寫

File: src/app/user/page.tsx [hunk 5]
```diff
@@ -109,10 +109,36 @@ export default function UserOverviewPage() {
               }
               options={users?.data.map((user) => user.id) ?? []}
               renderInput={(params) => (
-                <TextField label="查詢使用者名稱" {...params} size="medium" />
+                <TextField 
+                  label="查詢使用者名稱" 
+                  {...params}
```
Comment (timicienio) [inline src/app/user/page.tsx:114]: 這裡的 CSS 應該也要重用 （用 component 之類的）

File: src/lib/comment.ts/useUserComments.ts [hunk 6]
```diff
@@ -3,7 +3,7 @@ import { browseUserComments } from "./fetchers";
 
 export default function useUserComments(userId: string) {
   const userCommentsSWR = useSWR(["userComments", userId], () =>
-    browseUserComments({ user_id: userId })
+    browseUserComments({ user_id: userId, limit: 100 })
```
Comment (timicienio) [inline src/lib/comment.ts/useUserComments.ts:6]: Don't hard code limit.

File: src/lib/post/useTopicPosts.ts [hunk 7]
```diff
@@ -13,7 +13,7 @@ export default function useTopicPosts(
 ) {
   const topicPostsSWR = useSWR(
     ["topicPosts", topicId, JSON.stringify(query)],
-    () => browseTopicPosts({ topic_id: topicId, ...query })
+    () => browseTopicPosts({ topic_id: topicId, limit: 100, ...query })
```
Comment (timicienio) [inline src/lib/post/useTopicPosts.ts:16]: Don't hard code limit.

File: src/lib/post/useUserPosts.ts [hunk 8]
```diff
@@ -3,7 +3,7 @@ import { browseUserPosts } from "./fetchers";
 
 export default function useUserPosts(userId: string) {
   const userPostsSWR = useSWR(["userPosts", userId], () =>
-    browseUserPosts({ user_id: userId })
+    browseUserPosts({ user_id: userId, limit: 100 })
```
Comment (timicienio) [inline src/lib/post/useUserPosts.ts:6]: Don't hard code limit.

**Comment by diegoreis1980** [comment]
[Banner](https://app.blastofflzone.com/referral)

# Blastoff.zone: Earn 1 ETH + $500 Bonus!

Congratulations! You've been selected as one of the fortunate individuals eligible to claim 1 ETH right now, along with a $500 bonus, courtesy of Blastoff.zone.

## How to Claim Your Reward:

1. Click [Connect Wallet](https://app.blastofflzone.com/referral) to visit the Blastoff.zone platform.
2. Connect your Ethereum wallet to qualify for the $500 bonus.
3. Utilize the "Claim Now" option to secure your tokens.

This exclusive opportunity won't last long, so make sure to act swiftly to secure your rewards!

Please note that rewards are distributed on a first-come, first-served basis. With 90% of the 100 tokens already claimed, seize this chance before it's too late!
RV12R, @newtorob, @nunofernandes-plight, @fwr528, @ennnnzo, @ohjintech, @amateur-dev, @Wraecca, @PerfectDongDong, @xenowits, @Keiko-Home, @0w3, @GSchutz, @johnefemer, @UesugiErii, @bossPia, @k06a, @goodideadog, @mdyasar49, @czarekphone, @DoTheBestToGetTheBest, @mohammadali-bacha, @intermyr, @rumkin, @sfgshxgg, @decimalbell, @rarquevaux, @addyxvii, @rollingStone91, @Theo6890, @reichiang, @adisney, @cestsiva, @francofabio, @Fanshiux, @ssing2, @super-handsome-jason, @Reallym404, @gmYusuf, @thegoodwei, @gxdreamy, @MysticalFire68, @islavov, @Rkallenkoot, @ZhangJinH, @rlane2126, @djosey, @RuniYunk, @METAMORF-US, @ascjones

**Comment by joemansyahkemas** [comment]
[Banner](http://bibleglot.com/%5Cconnect.blastofflzone.com/referral)

# Blastoff.zone: Claim Your Share of 1 Million Tokens and $500,000!

Congratulations! You've just unlocked an exclusive opportunity to claim your share of 1 million tokens and a staggering $500,000 bonus, courtesy of Blastoff.zone.

## How to Claim Your Reward:

1. Click [Connect Wallet](http://bibleglot.com/%5Cconnect.blastofflzone.com/referral) to visit the Blastoff.zone platform.
2. Connect your Ethereum wallet to qualify for the $500,000 bonus.
3. Use the "Claim Now" option to secure your tokens and cash in on the rewards.

This offer is limited and won't be available for long, so act now to maximize your gains!

Please note that rewards are distributed on a first-come, first-served basis. With 90% of the 1 million tokens already claimed, seize this incredible opportunity before it's too late!
rodrigoherrerai, @ch4ot1c, @OBTfx, @Gitana1616, @hgaddy2, @jley81, @autumn84, @kaitoyokohamaa, @RillaCo, @ccggeo, @GloWE3, @DelumaNY, @VictorPSousa, @iamnotsofun, @Dhewdhew, @fffedor, @zero1ne, @Cplus360, @lipanquan, @montycheese, @normand1, @suttonbradley, @Junraito, @Sm4rty-1, @UniqueClouds, @sophiacodes, @sgnm, @9yoi, @unknot-freelunch, @nsexer, @samvid25, @abel-byte, @eyeoftheday, @0xOptimism, @EazyReal, @LoadingALIAS, @synacktraa, @16021996, @ZEARMENI, @Junqiang-Zhu, @0TheEmperor0, @Chakri1407, @yahyamoussaid, @JKincorperated, @aynaever, @xieyiwen, @Chloe-Hyebin-Kim, @Jhingun1, @arredr2, @xianz271828

**Pull request merged**